genie-client/src/main/java/com/netflix/genie/client/JobClient.java

```diff
@@ -461,34 +485,33 @@ public JobStatus waitForCompletion(final String jobId, final long blockTimeout,
                 return status;
             }
 
-            // block until timeout
-            try {
-                if (System.currentTimeMillis() - startTime < blockTimeout) {
-                    Thread.sleep(pollTime);
-                } else {
-                    throw new GenieClientException("Timed out waiting for job to finish");
-                }
-            } catch (InterruptedException ie) {
-                throw new IOException("Received interreupted exception from wait thread.");
+            if (System.currentTimeMillis() - startTime < blockTimeout) {
+                Thread.sleep(pollTime);
+            } else {
+                throw new GenieTimeoutException("Timed out waiting for job to finish");
             }
-
         }
     }
 
     /**
      * Wait for job to complete, until the given timeout.
      *
-     * @param id           the Genie job ID to wait for completion.
+     * @param jobId           the Genie job ID to wait for completion.
      * @param blockTimeout the time to block for (in ms), after which a
      *                     GenieClientException will be thrown.
      * @return The job status for the job after completion.
-     * @throws GenieClientException       For timeout or other errors.
-     * @throws InterruptedException on timeout/thread errors.
-     * @throws IOException If the response recieved is not 2xx.
+     *
+     * @throws InterruptedException on thread errors.
```

Comment:
Nitpick: I would probably reword this, as an InterruptedException does not necessarily indicate a "thread error". Here's what the Javadoc for Future.get says: "if the current thread was interrupted while waiting", which seems like a more accurate description.

I think the same Javadoc comment for InterruptedException is repeated elsewhere, so if we decide to change it, we should fix it everywhere.